modules/internal/ChapelDistribution.chpl

```diff
@@ -404,25 +404,44 @@ module ChapelDistribution {
 
       if !dataSorted && preserveInds {
         var _inds = inds;
-        return bulkAdd_help(_inds, dataSorted, isUnique); 
+        return bulkAdd_help(_inds, dataSorted, isUnique);
       }
       else {
         return bulkAdd_help(inds, dataSorted, isUnique);
       }
     }
 
-    proc bulkAdd_help(inds: [?indsDom] index(rank, idxType), 
+    proc bulkAdd_help(inds: [?indsDom] index(rank, idxType),
         dataSorted=false, isUnique=false){
       halt("Helper function called on the BaseSparseDomImpl");
 
       return -1;
     }
 
-    inline proc _grow(size: int){
+    // TODO: Would ChapelArray.resizeAllocRange() be too expensive?
+    //       - would have to put the conditional outside of the call
+    /* Grow domain if necesary */
+    inline proc _grow(size: int, factor=arrayAsVecGrowthFactor) {
```

Comment:
In hindsight, I may have gone overboard and these `_grow()` & `_shrink()` changes may be better fit for a follow-up PR with more thorough performance analysis. I assumed defaulting to what we do for array-resizing would be a more reasonable behavior than a hard-coded factor of `2`.